alot/commands/__init__.py

```diff
@@ -10,20 +10,53 @@
 import re
 
 from ..settings import settings
-from ..helper import split_commandstring, string_decode
+from ..helper import split_commandstring, string_decode, classproperty
 
 
 class Command(object):
 
     """base class for commands"""
     repeatable = False
+    mode = ""
+    cmdname = ""
+
+    @classproperty
+    def id(cls):
+        """Mode and command name of instance as string."""
+        return "{}.{}".format(cls.mode, cls.cmdname)
 
     def __init__(self):
-        self.prehook = None
-        self.posthook = None
+        # set to False if no attempt to fetch hooks from settings. After
+        # looking for hooks, they will contain a callable or None if no
+        # hook was found.
+        self._prehook = False
+        self._posthook = False
+
         self.undoable = False
         self.help = self.__doc__
 
+    @property
+    def prehook(self):
+        """Command pre hook."""
+        if self._prehook is False:
+            get_hook = settings.get_hook
+            mode = type(self).mode
+            cmdname = type(self).cmdname
+            self._prehook = (get_hook('pre_%s_%s' % (mode, cmdname)) or
+                             get_hook('pre_global_%s' % cmdname))
+        return self._prehook
```

Comment:
Is there a reason that the pre and post hooks should be applied to the instance rather than the class? I think that we actually want to attach these to the class.